cpp/src/plasma/client.cc

```diff
@@ -882,9 +882,10 @@ Status PlasmaClient::Impl::Disconnect() {
 
   // Close the connections to Plasma. The Plasma store will release the objects
   // that were in use by us when handling the SIGPIPE.
-  close(store_conn_);
-  store_conn_ = -1;
-  return Status::OK();
+  if (notification_conn_) {
+    RETURN_NOT_OK(notification_conn_->Disconnect());
```

Comment:
I'm not quite sure if we should return here. Maybe we should proceed to disconnect store_conn_ as well?